RemoteConnection#disconnect: fix ArgumentError on ruby 3.0

wowinter13 opened the pull request:
### Summary
ActionCable.server.broadcast raises ArgumentError in Ruby 3.0.0 when the message parameter is passed traditionally without using hashes.

Docs were fixed in https://github.com/rails/rails/pull/41050, but there is no fix for `RemoteConnection#disconnect`, which still uses an old syntax.

My env: 
```
$ rails -v
Rails 6.1.3
$ ruby -v
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin20]
```

Actual behavior:
```ruby
ActionCable.server.broadcast("sample_channel", type: "ping")
>> ArgumentError: wrong number of arguments (given 1, expected 2)

ActionCable.server.broadcast("sample_channel", { type: "ping" })
>> [ActionCable] Broadcasting to sample_channel: {:type=>"ping"}
```
 

Also check this related link: https://github.com/rails/rails/issues/40993

wowinter13 committed:
RemoteConnection#disconnect: fix ArgumentError on ruby 3.0

georgeclaghorn commented:
Seems like we’re missing tests covering `ActionCable::Server#disconnect` and `ActionCable::RemoteConnections`. Care to add some in this PR?

wowinter13 commented:
@georgeclaghorn Hello there
Yes, I saw this problem with tests but decided to propose a hotfix first of all.
Thanks, I will add specs in 1-2 days.

wowinter13 committed:
Fix docs on Server#broadcast and add specs for Server#disconnect

wowinter13 commented:
@georgeclaghorn Could you please check once again? I added a base test for `Server#disconnect` and fixed some docs with the old syntax. Talking about `RemoteConnections`, I will create another PR for this functionality (need to create more stubs). Would it be ok?

wowinter13 commented:
@georgeclaghorn ?

georgeclaghorn commented:
I don’t think we get a ton out of these heavily-stubbed tests. We’re verifying that the Action Cable internals are just-so instead of checking externally-observable effects.

I cherry-picked the fix in d2642762885d79b330930ea473357f965a0b5581. We can come back to the tests separately.

wowinter13 commented:
@georgeclaghorn Thanks!
